WireAuthentication/Sources/WireAuthenticationUI/Utilities/UniversalSheetModifier.swift

```diff
@@ -0,0 +1,89 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+import SwiftUI
+
+/// This ViewModifier creates a custom overlay for iPad and fallback to .sheet on iPhone
+struct UniversalSheetModifier<Item: Identifiable, SheetContent: View>: ViewModifier {
+
+    @Binding var item: Item?
+    var onDismiss: (() -> Void)?
+    var content: (Item) -> SheetContent
+
+    @ViewBuilder
+    func body(content: Content) -> some View {
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            content
+                .overlay {
+                    if let item {
+                        self.content(item)
+                            .applyPreferredSize()
+                    }
+                }
```

Comment:
we use a simple overlay, as on iPad .sheet is presented as a page and presentationDetents don't have any effect